src/libs/actions/IOU.js

```diff
@@ -122,82 +139,266 @@ function createIOUTransaction(params) {
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oneOnOneChatReport.reportID}`,
+                value: {
+                    ...oneOnOneCreatedReportAction,
+                    [oneOnOneIOUReportAction.sequenceNumber]: oneOnOneIOUReportAction,
+                },
+            },
+        );
+
+        successData.push(
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT}${oneOnOneChatReport.reportID}`,
+                value: {pendingFields: {createChat: null}},
+            },
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oneOnOneChatReport.reportID}`,
+                value: {
+                    0: {pendingAction: null},
+                    [oneOnOneIOUReportAction.sequenceNumber]: {pendingAction: null},
+                },
+            },
+        );
+
+        failureData.push(
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT}${oneOnOneChatReport.reportID}`,
+                value: {
+                    maxSequenceNumber: oneOnOneChatReportMaxSequenceNumber,
+                    lastReadSequenceNumber: oneOnOneChatReportMaxSequenceNumber,
+                    pendingFields: {createChat: null},
+                },
+            },
+            {
+                onyxMethod: CONST.ONYX.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oneOnOneChatReport.reportID}`,
+                value: {
+                    0: {pendingAction: null},
+                    [oneOnOneIOUReportAction.sequenceNumber]: {pendingAction: null},
+                },
+            },
+        );
+
+        splits.push({
+            email,
+            amount: splitAmount,
+            iouReportID: oneOnOneIOUReport.reportID,
+            chatReportID: oneOnOneChatReport.reportID,
+            transactionID: oneOnOneIOUReportAction.originalMessage.IOUTransactionID,
+            reportActionID: oneOnOneIOUReportAction.reportActionID,
+            clientID: oneOnOneIOUReportAction.clientID.toString(),
         });
+    });
+
+    return {
+        groupData: {
+            chatReportID: groupChatReport.reportID,
+            transactionID: groupIOUReportAction.originalMessage.IOUTransactionID,
+            reportActionID: groupIOUReportAction.reportActionID,
+            clientID: groupIOUReportAction.clientID.toString(),
+        },
+        splits,
+        onyxData: {optimisticData, successData, failureData},
+    };
 }
 
 /**
- * Creates IOUSplit Transaction for Group DM
- *
- * @param {Object} params
- * @param {Array} params.splits
- * @param {String} params.comment
- * @param {Number} params.amount
- * @param {String} params.currency
- * @param {String} params.reportID
+ * @param {Array} participants
+ * @param {String} currentUserLogin
+ * @param {Int} amount
```

Comment:
I think we use `Number` in most other cases. Or there is one case where `Integer` is used.

Reply:
I think Number is fine since we would say Float if it would be float :D